src/utils/getter.ts

```diff
@@ -0,0 +1,19 @@
+export function cacheObjectGetters<T, K extends PropertyKey = keyof T>(
+	object: T,
+	getterProperties: K[]
+) {
+	for (const property of getterProperties) {
+		const propertyGetter = Object.getOwnPropertyDescriptor(object, property)?.get;
+		if (propertyGetter) {
+			let cached: unknown;
+			Object.defineProperty(object, property, {
+				get: () => {
+					if (cached === undefined) {
+						cached = propertyGetter.call(object);
+					}
+					return cached;
+				}
+			});
```

Comment:
A stupid question: Wouldn't it make more sense to do the caching by replacing the
property with a non-getter fixed value on first call? That would also remove the overhead of the function call for subsequent calls:

```suggestion
			Object.defineProperty(object, property, {
				get: () => {
				  const value = propertyGetter.call(object);
				  // This replaces the getter with a fixed value for subsequent calls
				  Object.defineProperty(object, property, { value });
					return value;
				}
			});
```

Reply:
I think that works too. I was thinking that if it was originally a getter, it might be safer to keep it that way after too, but it seems like assigning a value after that still leaves it with `writable: false` so that's also fine.